### DISCUSSION TITLE
[Bug]: Unable to download files generated by the OpenAI assistants

### DISCUSSION THREAD

**Discussion by artemgordinskiy** [danny-avila/LibreChat#2633] [Troubleshooting] [answered]
### What happened?

When OpenAI assistants return a file, I'm unable to see or download them.
I am able to reliably reproduce this issue.
Running a fresh install using the latest `master` (`89b1e33be092e80abbc9522b6146519bff6226d2`)

### Steps to Reproduce

1. Create an assistant with the Code Interpreter capability
2. Give it a prompt to generate a graph, for instance
3. The code gets written but the file never shows up in the UI and is unavailable for download

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
LibreChat  | 2024-05-07 09:37:27 error: Error downloading file from OpenAI: 404 No such File object: file-vyA4ivhKGPlD4hezlXpAtOGN
LibreChat  | 2024-05-07 09:37:28 error: [/assistants/chat/] 404 No such File object: file-vyA4ivhKGPlD4hezlXpAtOGN
LibreChat  | 2024-05-07 09:39:42 error: [/assistants/chat/] Error cancelling run 400 Cannot cancel run with status 'cancelled'.
```

### Screenshots

![CleanShot 2024-05-07 at 13 24 19@2x](https://github.com/danny-avila/LibreChat/assets/1506254/469f234d-95c3-437e-9e4d-91e2fa586b48)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
weird, I can reliably not reproduce it: 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/646c30be-f8e6-411d-9236-a91b61e86788)

Are you using docker on a remote build, or locally? I'm wondering if that's part of the issue.

I'm having issues on my remote build with file links but not this exact issue.

**Reply by danny-avila** [reply]
actually my remote build just experienced a weird error on openai's side, it's also able to generate this fine:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/f8f9a043-7adf-4094-ba6c-7d64164c467a)

I can also view the images and download the file it generated

![image](https://github.com/danny-avila/LibreChat/assets/110412045/3273f0b6-f2f4-4924-8f3b-2b2265bb9c47)

**Reply by artemgordinskiy** [reply]
Interesting, thank you for looking into it!
Do you, perhaps, have some ideas on how I could debug the issue further on my side?

**Reply by danny-avila** [reply]
namely, how are you installing the app? file errors with the Assistants API could be due to a package mismatch, which can happen when using npm natively on your machine

**Reply by artemgordinskiy** [reply]
I installed it using Docker Compose on a Linux VPS, running behind Traefik. I didn't modify the default configuration pretty much at all.
Let me check the logs to see if there's anything suspicious there.

**Reply by danny-avila** [reply]
default docker compose is working for me, too. what about  if you ask a followup question, does it error out? are you also using azure or official OpenAI?

**Reply by danny-avila** [reply]
it's also possible the API key has read/write limitations.. i've not tried this yet though and can't confirm

**Reply by artemgordinskiy** [reply]
The only warnings I can see are these, but I'm not sure if they're relevant:

```
WARN[0000] The "UID" variable is not set. Defaulting to a blank string.
WARN[0000] The "GID" variable is not set. Defaulting to a blank string.
```

```
chat-meilisearch  | 2024-05-07T11:23:52.201325Z  WARN HTTP request{method=GET host="meilisearch:7700" route=/indexes/convos/documents/613e60a9-4d0c-470e-b2f6-c8274e7cb745 query_parameters= user_agent=node status_code=404 error=Document `613e60a9-4d0c-470e-b2f6-c8274e7cb745` not found.}: tracing_actix_web::middleware: Error encountered while processing the incoming HTTP request: ResponseError { code: 404, message: "Document `613e60a9-4d0c-470e-b2f6-c8274e7cb745` not found.", error_code: "document_not_found", error_type: "invalid_request", error_link: "https://docs.meilisearch.com/errors#document_not_found" }
chat-meilisearch  | 2024-05-07T11:23:52.201406Z  INFO HTTP request{method=GET host="meilisearch:7700" route=/indexes/convos/documents/613e60a9-4d0c-470e-b2f6-c8274e7cb745 query_parameters= user_agent=node status_code=404 error=Document `613e60a9-4d0c-470e-b2f6-c8274e7cb745` not found.}: meilisearch: close time.busy=199µs time.idle=108µs
```

**Reply by artemgordinskiy** [reply]
I'm using the official OpenAI API, using a key with full permissions

**Reply by danny-avila** [reply]
is it an assistant you had created before using the project/recently? I think the playground may not have backwards compat. with v1 and may be affecting it. I'll try this, too.

**Reply by artemgordinskiy** [reply]
No, I actually created the assistant today, using LibreChat 🤷‍♂️

**Reply by artemgordinskiy** [reply]
Interestingly, in cases like this (e.g. generating a PDF), it appears to work fine:
![CleanShot 2024-05-07 at 17 33 18](https://github.com/danny-avila/LibreChat/assets/1506254/03c4830e-7a6b-4511-8e22-55e01558fbdc)
So it's just image generation that consistently does not work 🤔

**Reply by danny-avila** [reply]
which model is it using? there are some recent bugs server-side on OpenAI's end: https://community.openai.com/t/invalid-model-gpt-3-5-does-not-support-image-message-content-types/728895/9

**Reply by artemgordinskiy** [reply]
Good question. So far, I tried 'gpt-4-turbo' and 'gpt-4-turbo-preview'. Should I try some others?

**Reply by artemgordinskiy** [reply, marked as answer]
@danny-avila Just found an accidental workaround:
Switched from a "Project API key" to a (legacy) "User API key" and the issue went away!

![CleanShot 2024-05-07 at 22 13 16@2x](https://github.com/danny-avila/LibreChat/assets/1506254/9342a133-7496-4a69-80ad-538f68d1636a)

Interestingly, both API keys had full access (e.g. all scopes enabled).
Also, tried multiple times for a good measure.

Do you think it's a bug in their API?

**Reply by danny-avila** [reply]
@artemgordinskiy great catch! Yes, they've had issues since the v1 -> v2 update for Assistants, lots of weird errors like this. I will be working on migrating this week to avoid other pitfalls like this, also there is no exact timeline over deprecation.

**Reply by mansiibm** [reply]
Hi can you please tell about azure open ai as well that how can i implement the above workaround?

**Reply by tommartens1997** [reply]
I have the same issue using azure open ai gpt 4o. Any news on this?

**Reply by danny-avila** [reply]
Please read about `retrievalModels`

https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/assistants_endpoint#retrievalmodels

We are adding `gpt-4o` as a recognized retrieval model, but the issue can be circumvented by the config right now
